app/Http/Controllers/PhotoController.php

```diff
@@ -108,11 +111,12 @@ public function add(AlbumIDRequest $request): Photo
 	/**
 	 * Change the title of a photo.
 	 *
-	 * @param Request $request
+	 * @param PhotoIDsRequest $request
+	 * @param SetTitle        $setTitle
 	 *
 	 * @return string
 	 */
-	public function setTitle(PhotoIDsRequest $request, SetTitle $setTitle)
+	public function setTitle(PhotoIDsRequest $request, SetTitle $setTitle): string
 	{
 		$request->validate(['title' => 'required|string|max:100']);
 
```

Comment:
don't you want to refactor that one similarly to the albums:
```php
		$setTitle->do(explode(',', $request['photoIDs']);

		return response()->noContent();
```

Reply:
At the moment: no. See [this comment](https://github.com/LycheeOrg/Lychee/pull/1101#discussion_r740435600) for an explanation. In the long run, yes. But this is another PR.